Sources/Sentry/SentryProfiler.mm

```diff
@@ -155,7 +143,30 @@
     }
 
     {
-        std::lock_guard<std::mutex> l(_gSamplesArrayLock);
+        // critical section: here is where all mutable data structures are updated, which must be
```

Comment:
`stacks` is still being mutated outside of the critical section on line 154

Reply:
Hm, for some reason I thought I wouldn't need to put the modification of `stacks` or `frames` here, but now I can't remember why, because we do put the modification for `samples` here and it's essentially the same structure. I'll add them.